lib/vrl/parser/src/parser.lalrpop

```diff
@@ -0,0 +1,511 @@
+    "else" => Ident("else".to_owned()),
+    "null" => Ident("null".to_owned()),
+    "true" => Ident("true".to_owned()),
+    "false" => Ident("false".to_owned()),
+};
+
+// -----------------------------------------------------------------------------
+// assignment
+// -----------------------------------------------------------------------------
+
+AssignmentExpr: Node<Expr> = {
+    Assignment => Node::new(<>.span(), Expr::Assignment(<>)),
+    ArithmeticExpr,
+};
+
+Assignment: Node<Assignment> = {
+    Sp<AssignmentSingle>,
+    Sp<AssignmentInfallible>,
+};
+
+AssignmentSingle: Assignment = {
+    <target: Sp<AssignmentTarget>> "="
+        <expr: Box<Expr>> => Assignment::Single { target, expr },
+}
+
+AssignmentInfallible: Assignment = {
+    <ok: Sp<AssignmentTarget>> ","
+        <err: Sp<AssignmentTarget>> "="
+        <expr: Box<Expr>> => Assignment::Infallible{ ok, err, expr},
+}
+
+#[inline]
+AssignmentTarget: AssignmentTarget = {
+    "_" => AssignmentTarget::Noop,
+    "." <Path?> => AssignmentTarget::External(<>),
+    Ident Path? => AssignmentTarget::Internal(<>),
+};
+
+// -----------------------------------------------------------------------------
+// arithmetic
+// -----------------------------------------------------------------------------
+
+ArithmeticExpr: Node<Expr> = {
+    Sp<ErrorCoalesce>,
+};
+
+ErrorCoalesce: Expr = {
+    Op<ErrorCoalesce, "??", Logical>,
+    Logical,
+};
+
+Logical: Expr = {
+    Op<Logical, "||", Equal>,
+    Op<Logical, "&&", Equal>,
+    Equal,
+};
+
+Equal: Expr = {
+    Op<Equal, "!=", Compare>,
+    Op<Equal, "==", Compare>,
+    Compare,
+};
+
+Compare: Expr = {
+    Op<Compare, ">=", Add>,
+    Op<Compare, ">", Add>,
+    Op<Compare, "<=", Add>,
+    Op<Compare, "<", Add>,
+    Add,
+};
+
+Add: Expr = {
+    Op<Add, "+", Factor>,
+    Op<Add, "-", Factor>,
+    Factor,
+};
+
+Factor: Expr = {
+    Op<Factor, "%", Not>,
+    Op<Factor, "*", Not>,
+    Op<Factor, "/", Not>,
+    Not,
+};
+
+Not: Expr = {
+    <bang: Sp<"!">> <expr: Box<Sp<Not>>> => {
+        let span = span(bang.span().start(), expr.span().end());
+        let not = Unary::Not(Node::new(span, Not(bang.map(|_| ()), expr)));
+        Expr::Unary(Node::new(span, not))
+    },
+    Term,
+};
+
+Term: Expr = {
+    Sp<Literal> => Expr::Literal(<>),
+    Sp<Container> => Expr::Container(<>),
+    Sp<Query> => Expr::Query(<>),
+    Sp<FunctionCall> => Expr::FunctionCall(<>),
+    Sp<Ident> => Expr::Variable(<>),
+};
+
+// -----------------------------------------------------------------------------
+// query
+// -----------------------------------------------------------------------------
+
+pub Query: Query = {
+    LQuery <Sp<".">> RQuery => {
+        let span = <>.span();
+        let target = Node::new(span, QueryTarget::External);
+        let path = Node::new(span, Path(vec![]));
+
+        Query { target, path }
+    },
+    LQuery <target: Sp<QueryTarget>> <path: Sp<Path>> RQuery => Query { target, path },
+};
+
```

Comment:
There is an issue here. I can't quite work out what it is. But this doesn't parse - `.a == 3`, nor does `.a + 3 == 5`.

Interestingly `.a + 3` does, as does `a == 3`.

Reply:
Good catch. Fixed locally.